### PR TITLE
Station Refresh

### LINKED ISSUE #3: QR code for WhatsApp does not generate
I wanted to pair the Station with WhatsApp but I cannot get the QR code, there is this circle moving. I restarted the Station, reset the app (WhatsApp), reloaded, did not help
wa

I have macOS 11.5 (the latest public beta)

### LINKED ISSUE #18: The new version is not loading at the Linux (Elementary OS distro)
When I try to run the AppImage at the terminal, is just saying that the getName function is deprecated:

```
~$ ./Downloads/Station-x86_64.AppImage 
(electron) 'getName function' is deprecated and will be removed. Please use 'name property' instead.
^C^C
```

### LINKED ISSUE #24: AppImage can't create Telegram QR Code
The AppImage version
- Stationversion 2.0.9
- Electron7.3.3
- Chrome78.0.3904.130
- Node12.8.1
- v87.8.279.23-electron.0
The Telegram in QR Code look like this:
圖片

It can't generate a QR Code.

### LINKED ISSUE #27: QR Code - not showing
Hello, on macbook pro M1, QRcodes for login in to whatsapp, telegram not showing and cannot connect. the QR is only with loading icon.
image
image

### LINKED ISSUE #28: Unable to add google account
Tried to connect google account on the latest desktop app on Mac. I get the below error. 

image

### LINKED ISSUE #29: Cannot use with multiple google accounts
Like others have mentioned, this has been my main app to handle multiple google accounts in mail, calendar and contacts. However, trying the workaround offered in this issue #2 with a customized app, I can only login to one google account at a time unlike the soon-to-be deprecated version.

Is there any update on whether the google authentication issues are prioritized or not?! I would hate to give up on an otherwise amazing tool!

### LINKED ISSUE #30: Any news about WhatsApp and Telegram ???
Hello guys ... 
Please, any news about bugs on Telegram, WhatsApp (and GMail ... I still use the workaround with the url to the gmail page)?

But my problem is with Telegram and WhatsApp ... I can not use them at all ... QR-Code is NEVER shown and I can not enter them.

Many thanks.
Pietro

### LINKED ISSUE #40: I can't login to google and get an error message.
The contents of the error message are as follows.
How can I do this?

Authorization Error
Error 400: invalid_request
Missing required parameter: client_id

### LINKED ISSUE #73: Whatsapp QR code keeps loading
Hi all,

The WhatsApp QR code keeps loading, I can't scan it to activate on my Station app.
Does anyone have the same issue?

### LINKED ISSUE #82: AppImage Fails to start
### Process

1. Download [AppImage](https://github.com/getstation/desktop-app/releases/latest/download/Station-x86_64.AppImage) 
2. mv Download/Station-x86_64.AppImage /opt
3. chmod +x Station-x86_64.AppImage
4. ./Station-x86_64.AppImage

#### My System

- Fedora 35 (Workstation)
- Kernel: 5.14.16-301.fc35.x86_64
- Gnome version 41.1
- Running X11
   - version 11.0
   - X.Org version: 1.20.11

#### Error When running App Image
```bash
./Station-x86_64.AppImage
(electron) 'getName function' is deprecated and will be removed. Please use 'name property' instead.
Fontconfig warning: "/usr/share/fontconfig/conf.avail/05-reset-dirs-sample.conf", line 6: unknown element "reset-dirs"
[23040:1108/115844.469867:FATAL:gpu_data_manager_impl_private.cc(986)] The display compositor is frequently crashing. Goodbye.
[1]    23040 trace trap (core dumped)  ./Station-x86_64.AppImage
```

### LINKED ISSUE #93: Unable to login in Telegram
Unable to login in Telegram, Infinity loading and can't loading QR Code

### LINKED ISSUE #111: Native M1 Binaries - Supported?
I tried installing Station but it seems that it's not provide native m1 binaries yet. will this be supported in the future?

### LINKED ISSUE #114: Gmail not working
I am a new user here, I tried to add my Gmail account to the station and this is not working, can you please help me out? 

PS - I tried to copy link into Chrome, still no good

Screen Shot 2022-03-21 at 8 51 57 AM

### LINKED ISSUE #115: Slack is not working on current version.
Adding an option to change user-agent by app could fix this.

### LINKED ISSUE #116: Can't configure WhatsApp - no QR Code - version 2.0.9
As in subject - I can't configure WhatsApp on fresh Station (and fresh Windows 10) installation. App doesn't show qr code , it shows only spinning circle. Neverending spinning. What to do?

### LINKED ISSUE #117: Gmail not working and whatsapp not showing the QR code
Hi! 
When I try to add my Gmail account it shows the following error: Erro 400: invalid_request
Missing required parameter: client_id

Whatsapp is ln an infinite QR code lead and it doesn't appear even after quitting and opening the app again. 

Can someone give me any tips? 
Telegram worked fine!

### LINKED ISSUE #120: Update Browser Version — Chrome? I think
Is there any way to update the browser that Station uses? For example, when i log into telegram, it says it's logged in Chrome 78, which has quite a lot of security vulnerabilities. When I also go to whatismybrowser.com through Station, it shows me that chrome is version 78. Is there a workaround to update chrome?

Screen Shot 2022-04-18 at 9 05 09 AM

### LINKED ISSUE #124: 400 Authorization Error when connecting to GMail
Gmail App. Clicking on Add a new account opens a browser. 
I enter sign in details. I get google verification. 
Then it shows 400 Authorization Error.
I tried switching from Chrome to Firefox, same 400 Auth Error.

### LINKED ISSUE #129: /macOS/ no qr code shown for whatsapp and telegram? chrome updated to the last version.
as per the title it seems like I can't see the qr code to configure whatsapp and telegram. any help?

### LINKED ISSUE #130: LinkedIn App Hanging at Loading Bar
Linkedin In app fails to load, hanging on loading bar
Screenshot 2022-07-21 at 15 27 32
.

### LINKED ISSUE #132: Slack Error - We're very sorry, but your browser is not supported!
This is the message that I get when I add Slack to Station and enter my workspace address.

We're very sorry, but your browser is not supported!

I am using Windows 11 and the latest version of Edge as my default browser.

### LINKED ISSUE #134: Adding Gmail account throws error
When trying to add gmail account, when logging into account I get "missing client_id".

I guess something is wrong?

### LINKED ISSUE #137: Authorization Error
im trying to connect gmail for the first time and i get this when i login 

Authorization Error
Error 400: invalid_request
Missing required parameter: client_id

any idea whats going on?

### LINKED ISSUE #138: Still can install on Linux (Fedora 36) ?

### LINKED ISSUE #140: Google Calendar login invalid_request
Downloaded. Selected Google Calendar.  Selected "Add a new account". Redirected to browser to log in to google. Onced logged in:
Missing required parameter: client_id
400: invalid_request
image

### LINKED ISSUE #142: Can't login to whatsapp
the QR code not showing, Just Loading 
edit: after a while I got error, asking me to update chrome
image

### LINKED ISSUE #144: Asana incompatible browser.
I cannot connect to Asana both on Macbook and Windows. The browser seems incompatible, I tried the recommended browser and nothing worked.

### LINKED ISSUE #147: Error while opening station app
Getting the following error while opening up station app. Here's the neofetch result for your reference.

**Error Message**
station(electron) 'getName function' is deprecated and will be removed. Please use 'name property' instead.

(station:16533): Gtk-WARNING **: 14:00:56.837: Theme parsing error: gtk.css:73:46: The style property GtkScrolledWindow:scrollbars-within-bevel is deprecated and shouldn't be used anymore. It will be removed in a future version
Gtk-Message: 14:00:56.853: Failed to load module "appmenu-gtk-module"
Fontconfig warning: "/usr/share/fontconfig/conf.avail/05-reset-dirs-sample.conf", line 6: unknown element "reset-dirs"
[16533:0430/140059.898922:FATAL:gpu_data_manager_impl_private.cc(986)] The display compositor is frequently crashing. Goodbye.
[1]    16533 trace trap (core dumped)  station

**Neofetch Results**
OS: Manjaro Linux x86_64 
Host: B150M-D3H 
Kernel: 6.2.12-1-MANJARO 
Uptime: 1 hour, 56 mins 
Packages: 1510 (pacman), 10 (flatpak), 7 (snap) 
Shell: zsh 5.9 
Resolution: 1920x1080 
DE: Plasma 5.27.4 
WM: KWin 
WM Theme: Oxygen 
Theme: Breeze Dark [Plasma], Breath-Dark [GTK2/3] 
Icons: oxygen [Plasma], oxygen [GTK2/3] 
Terminal: terminator 
CPU: Intel i5-6500 (4) @ 3.600GHz 
GPU: NVIDIA GeForce GT 730 
Memory: 3786MiB / 15955MiB

### PR DESCRIPTION
**Pull request by viktor44**
Major changes:
* Electron was upgraded up to 13.6.9
* Google Legacy People API was replaced with modern one

Issues:
* Fixes #147
* Fixes #144
* Fixes #142
* Fixes #140
* Fixes #138
* Fixes #137
* Fixes #134
* Fixes #132
* Fixes #130
* Fixes #129
* Fixes #124
* Fixes #120
* Fixes #117
* Fixes #116
* Fixes #115
* Fixes #114
* Fixes #111
* Fixes #93
* Fixes #82
* Fixes #73
* Fixes #40
* Fixes #30
* Fixes #29
* Fixes #28
* Fixes #27
* Fixes #24
* Fixes #18
* Fixes #3

Pre-releases here https://github.com/viktor44/station-desktop-app/releases

### CODE REVIEW

File: app/session.ts [hunk 1]
```diff
@@ -1,105 +1,139 @@
...
-const callbackMethods = [
-  'onBeforeRequest',
-  'onBeforeSendHeaders',
-  'onHeadersReceived',
-];
+// const callbackMethods = [
+//   'onBeforeRequest',
+//   'onBeforeSendHeaders',
+//   'onHeadersReceived',
+// ];
```
Comment (magne4000) [inline app/session.ts:60]: Could you add a comment explaining why this is commented?
Something like the following should do:

<pre>
Code related to Chrome Extensions, which are (currently) not supported
</pre>

File: app/session.ts [hunk 2]
```diff
@@ -1,105 +1,139 @@
...
+  'Safari/537.36',
+].join(' ');
 
 const whatsappUserAgent = [
   'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_1)',
   'AppleWebKit/537.36 (KHTML, like Gecko)',
-  'Chrome/76.0.3809.100 Safari/537.36',
+  'Chrome/76.0.3809.100',
+  'Safari/537.36',
 ].join(' ');
```
Comment (magne4000) [inline app/session.ts:74]: With the new default UA, this one is probably no relevant anymore

**Review by magne4000 — APPROVED** [review]
:rocket: Awesome job @viktor44!

I'm preparing a release with this PR right now

File: app/session.ts (hunk 2 above)
Comment (viktor44) [reply app/session.ts:74]: It works, so I decided not to change it.

Comment (magne4000) [reply app/session.ts:74]: Good for me :)

File: app/session.ts (hunk 1 above)
Comment (viktor44) [reply app/session.ts:60]: It looks like this hack

https://github.com/getstation/desktop-app/blob/f4c2de9c1c7e44cc0e522e7e19a518134f2fa088/app/session.ts#L94-L104

doesn't work anymore.

So I replaced it with [this](https://github.com/viktor44/station-desktop-app/blob/400360863dd1068e6eee484853c271524309308b/app/main.ts#L81-L90)

```javascript
const initWorker = () => {
  app.on('ready', () => {

    session.defaultSession.webRequest.onBeforeSendHeaders((details: OnBeforeSendHeadersListenerDetails, callback: any) => {
      details.requestHeaders['User-Agent'] = getUserAgentForApp(details.url, session.defaultSession.getUserAgent());
      details.referrer = getRefererForApp(details.referrer);
      details.requestHeaders.Referer = details.referrer;

      callback({ cancel: false, requestHeaders: details.requestHeaders });
    });
```

**Comment by viktor44** [comment]
@magne4000  It is necessary to update `node-gyp` that goes with `npm` up to `node-gyp@8.4.1`
Otherwise it fails.
I haven't yet updated `readme.md`

**Comment by magne4000** [comment]
[Windows build fails](https://github.com/getstation/desktop-app/actions/runs/4891609186/jobs/8732335323?pr=148) because of `windows-quiet-hours`.
@viktor44 I do not have a Windows quickly available to try to fix that, and I see that you did manage to create a Windows build. Could you take a look at it?

**Comment by magne4000** [comment]
> @magne4000 It is necessary to update `node-gyp` that goes with `npm` up to `node-gyp@8.4.1` Otherwise it fails. I haven't yet updated `readme.md`

`yarn.lock` does have `node-gyp@8.4.1` set, so it should be OK

**Comment by magne4000** [comment]
Oh no you're right, I just saw see that `gyp ERR! node-gyp -v v5.1.1`...

**Comment by viktor44** [comment]
I updated `node-gyp` version on every OS
You have to run `npm i node-gyp@8.4.1` from those dirs:
* Linux - _I'll check later_
* Windows - _I'll check later_
* MacOS - `~/.nvm/versions/node/v14.21.3/lib/node_modules/npm`

**Comment by magne4000** [comment]
@viktor44 I added you as a maintainer to the repo, so you should be able to run the CI

**Comment by magne4000** [comment]
One issue remains: tests are failing on Windows because the migrations script doesn't seem to run, but it's not obvious why.

**Comment by magne4000** [comment]
@viktor44 great work, thanks again!

**Comment by magne4000** [comment]
:rocket: https://github.com/getstation/desktop-app/releases/tag/v2.1.0

**Comment by katastrophe92** [comment]
Hi. It is now working like before? Are any Beta releases?
Big thanks @viktor44 for reviewing this project.

**Pull request merged**